examples/light-switch-app/nrfconnect/main/AppTask.cpp

```diff
@@ -0,0 +1,601 @@
+    }
+
+    /* Initialize UI components */
+    LEDWidget::InitGpio();
+    LEDWidget::SetStateUpdateCallback(LEDStateUpdateHandler);
+    sStatusLED.Init(DK_LED1);
+    sBleLED.Init(DK_LED2);
+    sDiscoveryLED.Init(DK_LED3);
+    sUnsedLED.Init(DK_LED4);
+    UpdateStatusLED();
+
+    int ret = dk_buttons_init(ButtonEventHandler);
+
+    if (ret)
+    {
+        LOG_ERR("dk_buttons_init() failed");
+        return chip::System::MapErrorZephyr(ret);
+    }
+
+    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+
+    /* Initialize DFU */
+
+#ifdef CONFIG_MCUMGR_SMP_BT
+    /* Initialize DFU over SMP */
+    GetDFUOverSMP().Init(RequestSMPAdvertisingStart);
+    GetDFUOverSMP().ConfirmNewImage();
+#endif
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+    sOTAImageProcessor.SetOTADownloader(&sBDXDownloader);
+    sBDXDownloader.SetImageProcessorDelegate(&sOTAImageProcessor);
+    sOTARequestorDriver.Init(&sOTARequestor, &sOTAImageProcessor);
+    sRequestorStorage.Init(Server::GetInstance().GetPersistentStorage());
+    sOTARequestor.Init(Server::GetInstance(), sRequestorStorage, sOTARequestorDriver, sBDXDownloader);
+    chip::SetRequestorInstance(&sOTARequestor);
+#endif
+
+    /* Initialize Timers */
+    k_timer_init(&sFunctionTimer, AppTask::TimerEventHandler, nullptr);
+    k_timer_init(&sDimmerPressKeyTimer, AppTask::TimerEventHandler, nullptr);
+    k_timer_init(&sDimmerTimer, AppTask::TimerEventHandler, nullptr);
+    k_timer_user_data_set(&sDimmerTimer, this);
+    k_timer_user_data_set(&sDimmerPressKeyTimer, this);
+    k_timer_user_data_set(&sFunctionTimer, this);
+
+    /* Print initial configs */
+    ReturnErrorOnFailure(chip::Server::GetInstance().Init());
+    ConfigurationMgr().LogDeviceConfig();
+    PrintOnboardingCodes(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
+
+    /*
+     * Add CHIP aEvent handler and start CHIP thread.
+     * Note that all the initialization code should happen prior to this point
+     * to avoid data races between the main and the CHIP threads.
+     */
+    PlatformMgr().AddEventHandler(ChipEventHandler, 0);
+
+    err = PlatformMgr().StartEventLoopTask();
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("PlatformMgr().StartEventLoopTask() failed");
+        return err;
+    }
+
+    LOG_INF("Init CHIP stack done...");
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR AppTask::StartApp()
+{
+    ReturnErrorOnFailure(Init());
+
+    AppEvent aEvent = {};
+
+    while (true)
+    {
+        k_msgq_get(&sAppEventQueue, &aEvent, K_FOREVER);
+        DispatchEvent(aEvent);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void AppTask::PostEvent(const AppEvent & aEvent)
+{
+    if (k_msgq_put(&sAppEventQueue, &aEvent, K_NO_WAIT))
+    {
+        LOG_INF("Failed to post aEvent to app task aEvent queue");
+    }
+}
+
+void AppTask::DispatchEvent(const AppEvent & aEvent)
+{
+    switch (aEvent.Type)
+    {
+    case AppEvent::FunctionButtonPress:
```

Comment:
Maybe we can think about replacing such large switches with map/dict concept.

Reply:
Hmm... it's good point, I wanted to be in line with lock-app and lighting-app examples, but I agree with that switch here is too extensive.

Reply:
@kkasperczyk-no @Damian-Nordic what do you think about it?

Reply:
Sure, I would rather suggest that we create a separate task/spike to at least consider `utils` implementation for all our examples to actually limit the maintenance cost. That could include for instance: unified chip/OTA initialization and some utils to handle app events more cleanly (like aforementioned map etc. - I believe we are very much limited in Matter as far as STL is concerned, right?)